File: 3_module/VisMealy/VisMealy.go
package main

import "fmt"

type state struct {
	in  int
	out string
}

func main() {
	var n, m, q_start int
	fmt.Scanf("%d\n%d\n%d\n", &n, &m, &q_start)
	M := make([][]state, n)
	for i := 0; i < n; i++ {
		M[i] = make([]state, m)
	}
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			fmt.Scanf("%d", &M[i][j].in)
		}
		fmt.Scanf("\n")
	}
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			fmt.Scanf("%s", &M[i][j].out)
		}
		fmt.Scanf("\n")
	}
	fmt.Printf("digraph {\n\trankdir = LR\n")
	//	for i := 0; i < n; i++ {
	//		fmt.Printf("\t%d [shape = circle]\n", i)
	//	}
	//	fmt.Printf("\tdummy -> %d\n", q_start)
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			fmt.Printf("\t%d -> %d [label = \"%c(%s)\"]\n", i, M[i][j].in, 'a'+j, M[i][j].out)
		}
	}
	fmt.Printf("}")
}

File: 3_module/VisMealy/VisMealy.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int n, m, q;
    cin >> n >> m >> q;
    vector <vector <int> > gm (n, vector <int> (m));
    vector <vector <string> > em (n, vector <string> (m));
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            cin >> gm[i][j];
        }
    }
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            cin >> em[i][j];
        }
    }
    cout << "digraph{" << endl << "\trankdir = LR" << endl;
    for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
            cout << "\t"  << i << " -> " << gm[i][j] << " [label = \"" << char(97+j) << "(" << em[i][j] << ")\"]\n";
			//printf("\t%d -> %d [label = \"%c(%c)\"]\n", i, gm[i][j], alph[j], em[i][j]);
		}
		q++;
	}
	cout << "}";
}
